luarules/gadgets/unit_prevent_cloaked_unit_reclaim.lua

```diff
@@ -36,22 +39,28 @@ if gadgetHandler:IsSyncedCode() then
     end
     
     function gadget:AllowUnitCloak(unitID) -- cancel reclaim commands
-        local x, y, z = GetUnitPosition(unitID)
-        local units = GetUnitsInCylinder(x, z, maxBuildDist)
-        for _, bID in pairs(units) do
-            local unitDefID = GetUnitDefID(bID)
-            if canReclaim[unitDefID] then
-                local cmd, target = GetUnitWorkerTask(bID)
-                if cmd == CMD.RECLAIM and target == unitID and (GetUnitAllyTeam(bID) ~= GetUnitAllyTeam(unitID)) and not IsUnitInRadar(unitID, GetUnitAllyTeam(bID)) then
-                    local _, _, cmdTag = GetUnitCurrentCommand(bID, 1)
-                    GiveOrderToUnit(bID, CMD.REMOVE, {cmdTag}, {})
+        if (cloakedUnits[unitID]) and (not checkedUnits[unitID]) then  -- only needs to be checked when the unit barely is cloaked
+            checkedUnits[unitID] = true
```

Comment:
why not just do it in UnitCloaked?

Reply:
It needs to be issued later, because it is possible to re-issue the command in the last second before the unit cloaks(like when nanos are on fight). `cloakedUnits[unitID]` checks for that it has waited enough, and `checkedUnits[unitID]` improves performance by only checking right after the unit cloaks.

Reply:
Can this be resolved?